### ISSUE TITLE
[Android Bootcamp] Project 0: Review my todo app

### ISSUE THREAD

**Issue by urspralini** [urspralini/codepath-todoList#5]
Todo app is complete with optional sections: 
1. Used ActiveAndroid to store todo items in SQLlite.
2. Used custom adapter and implemented view holder to customize the view. (right now, it is only a text view).
Please review /cc @codepathreview @codepath

**Comment by nesquena** [comment]
Prabhu,

Looks good, this exercise is intended in part to give you an introduction to the general rhythm of this course. The course is entirely project-based with an app being assigned each week and then due the following week. Each project builds on the last to help each engineer  learn the practical elements of Android development and best practices as quickly as possible. We also do a code review for each submitted project once the program begins.

Great to see you were able to complete some extension tasks already as well. The extension tasks available on each project are often the most valuable learnings since they dive deeper into common real-world Android use cases.

The next step is to continue working on [extensions to your todo app](http://courses.codepath.com/snippets/intro_to_android/prework#heading-5-extending-your-todo) and to **schedule a short 10 minute phone conversation** with us. Navigate to [today's date and onward from there](https://calendar.google.com/calendar/selfsched?sstoken=UURvT01vbzBza1pnfGRlZmF1bHR8NzVjYzBkN2EyZTRiOGRmYjZlZjhmMjA2N2IyMGMwYWU) and choose an available 15-minute slot in the mornings or evenings. Let us know if none of those times work.

Once you select a slot, can you make sure to **include the best number** to reach you at in the invite? Look forward to chatting soon!

**Comment by urspralini** [comment]
Hi Nathan (@nesquena),

Good afternoon. Yes, I'm currently working on
the view customization with more fields like date-picker, status and priority drop downs. I'm
very much interested in participating in code path program and learning
Android programming. I have scheduled the phone call conversation for
6/23(Thursday) at 11.15 am. I would like to know what I should be expecting
in that phone conversation. It would be helpful if I should do any
preparation before that.

Thanks,
Prabhu

On Mon, Jun 13, 2016 at 7:48 PM, Nathan Esquenazi notifications@github.com
wrote:

> Prabhu,
> 
> Looks good, this exercise is intended in part to give you an introduction
> to the general rhythm of this course. The course is entirely project-based
> with an app being assigned each week and then due the following week. Each
> project builds on the last to help each engineer learn the practical
> elements of Android development and best practices as quickly as possible.
> We also do a code review for each submitted project once the program begins.
> 
> Great to see you were able to complete some extension tasks already as
> well. The extension tasks available on each project are often the most
> valuable learnings since they dive deeper into common real-world Android
> use cases.
> 
> The next step is to continue working on extensions to your todo app
> http://courses.codepath.com/snippets/intro_to_android/prework#heading-5-extending-your-todo
> and to _schedule a short 10 minute phone conversation_ with us. Navigate
> to today's date and onward from there
> https://calendar.google.com/calendar/selfsched?sstoken=UURvT01vbzBza1pnfGRlZmF1bHR8NzVjYzBkN2EyZTRiOGRmYjZlZjhmMjA2N2IyMGMwYWU
> and choose an available 15-minute slot in the mornings or evenings. Let us
> know if none of those times work.
> 
> Once you select a slot, can you make sure to _include the best number_ to
> reach you at in the invite? Look forward to chatting soon!
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/urspralini/codepath-todoList/issues/5#issuecomment-225766112,
> or mute the thread
> https://github.com/notifications/unsubscribe/AE7-OhKg-tlKo_kFjot6GvVDN8knhtf5ks5qLhabgaJpZM4I0wVv
> .

## 

-Prabhu

## 

Legal Disclaimer: The information contained in this message may be 
privileged and confidential. It is intended to be read only by the 
individual or entity to whom it is addressed or by their designee. If the 
reader of this message is not the intended recipient, you are on notice 
that any distribution of this message, in any form, is strictly prohibited. 
If you have received this message in error, please immediately notify the 
sender and delete or destroy any copy of this message

**Comment by nesquena** [comment]
> I have scheduled the phone call conversation for
> 6/23(Thursday) at 11.15 am. I would like to know what I should be expecting
> in that phone conversation. It would be helpful if I should do any
> preparation before that.

Sorry for the delay here. We will be discussing your goals for the course, the course expectations and I'll answer any questions you have about the course. If you have time before our interview tomorrow can you take a little time to research the most common Android Activity lifecycle methods as well as the `ArrayAdapter` pattern used when working with lists? We will be discussing these with you conceptually in the phone call. We encourage you to [search our guides](http://guides.codepath.com/android/) as well as any other sources. If not, no worries.

**Comment by urspralini** [comment]
Thanks @nesquena for the detailed reply. I'll learn about Activity life cycle methods and ArrayAdapter before our call. Our call is scheduled for next Thursday(6/23) at 11.15am.

**Comment by nesquena** [comment]
Great, I'll chat with you then!

**Comment by urspralini** [comment]
Todo app is updated with following optional sections:
- Add support for completion due dates for todo items (and display within listview item)
- Add support for selecting the priority of each todo item (and display in listview item)
- Tweak the style improving the UI / UX, play with colors, images or backgrounds.

Please review /cc @codepathreview @codepath

**Comment by nesquena** [comment]
Great, thanks for the update and glad to see you got a number of optionals implemented. We will be determining final admittance into the program over the next couple weeks. For those not admitted as participants, we will offer access to applicants as a remote observer cohort that has access to the materials and is able to work on the projects at their own pace. You will be hearing from us again soon.

**Comment by urspralini** [comment]
thanks @nesquena for your reply. Todo app is updated with following optional sections:
- Using Alert Dialog (Dialog fragment) to confirm the deletion of an todo item.
- Added "sort by" feature to sort the list view items based on fields such as Priority, Status, Due Date and Name.
- Used TypeSerializer in active android to custom serialize the values of Status and Priority. This helps in sorting feature. 

Please review /cc @codepathreview @codepath

**Comment by nesquena** [comment]
Great, great job with the pre-work. Hope you feel you were able to learn a lot working through these optionals.

**Comment by urspralini** [comment]
Yes @nesquena. I feel comfortable using the "active android" library now. I also learnt about passing values between activities and dialog fragment from the first project. I still need to get better with UI design: views, layouts, positioning, margin, padding, custom shapes etc. :).
Looking forward for the results soon.
